xcode/Subconscious/Shared/Services/Noosphere/NoosphereService.swift

```diff
@@ -443,6 +445,40 @@ actor NoosphereService:
         try await self.sphere().verify(authorization: authorization)
     }
     
+    func recover(identity: Did, gatewayUrl: URL, mnemonic: RecoveryPhrase) async throws -> Bool {
+        // Update the gateway URL to whatever was in the form
+        resetGateway(url: gatewayUrl)
+        // Release the sphere before we attempt to recover it
+        // If we don't do this the database LOCK will prevent us from recovering
+        resetSphere(nil)
+        
+        let noosphere = try noosphere()
+        
+        let result: Bool = try await withCheckedThrowingContinuation { continuation in
```

Comment:
It feels like this portion of the code should belong to `Noosphere`, since `NoosphereService` is ultimately meant to be a thin memoization layer on top of `Noosphere`.

Reply:
Agreed. Moved into Noosphere.